### PR TITLE
Runs crucible for sql

### PR DESCRIPTION
**Pull request by rbans96**
## Description
Adds crucible support for SQL

## Related issues
Addresses : User Story 70320

## Testing
Run crucible tests locally
image

### CODE REVIEW

**Comment by brendankowitz** [comment]
I'm pretty sure you'll need to set something inside `SetTestServerAsync` so that the test server names are something like "PR123_R4_SQL"

**Comment by brendankowitz** [comment]
The crucible server is: https://crucible.mshapis.com/ you should be able to see both variations show up after the E2E tests run

**Comment by rbans96** [comment]
> The crucible server is: https://crucible.mshapis.com/ you should be able to see both variations show up after the E2E tests run

yes I can see that :)
image

File: test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleDataSource.cs [hunk 1]
```diff
@@ -88,5 +90,10 @@ private async Task<ServerTestRun> GetTestRunAsync()
 
             return new ServerTestRun(crucible.ServerBase, result);
         }
+
+        public static DataStore GetDataStore()
+        {
+            return _dataStore;
+        }
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleDataSource.cs]: This looks like it should be a property

File: test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleTestFixture.cs [hunk 2]
```diff
@@ -37,12 +40,14 @@ public void Run(string id)
 
             if (findTest != null)
             {
+                bool isCosmosDb = CrucibleDataSource.GetDataStore().Equals(DataStore.CosmosDb);
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleTestFixture.cs]: In general we prefer to use string.Equals and also pass in the type of string comparison, so for example here: `string.Equals(a, b, StringComparison.OrdinalIgnoreCase)`

File: test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleTestFixture.cs [hunk 3]
```diff
@@ -37,12 +40,14 @@ public void Run(string id)
...
                 var failures = findTest.Result
                     .Where(x =>
                     {
                         var testName = $"{x.TestId ?? findTest.TestId}/{x.Id}";
-                        return x.Status == "fail" && !KnownCrucibleTests.KnownFailures.Contains(testName) && !KnownCrucibleTests.KnownBroken.Contains(testName)
+                        bool commonFailures = x.Status == "fail" && !KnownCrucibleTests.KnownCommonFailures.Contains(testName) && !KnownCrucibleTests.KnownBroken.Contains(testName)
                                && !x.Message.ToString().Contains(KnownCrucibleTests.BundleCountFilter);
+                        return isCosmosDb ? commonFailures && !KnownCrucibleTests.KnownCosmosDbFailures.Contains(testName) : commonFailures && !KnownCrucibleTests.KnownSqlServerFailures.Contains(testName);
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleTestFixture.cs]: As a suggestion we might be able to simplify by creating a `knownFailures` variable with this logic built-in to use here and below.
e.g. `var knownFailures = commonFailures.Concat(isCosmosDb ? KnownCrucibleTests.KnownCosmosDbFailures : KnownCrucibleTests.KnownSqlServerFailures).ToArray();`

File: test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleDataSource.cs (hunk 1 above)
Comment (rbans96) [reply test/Microsoft.Health.Fhir.Shared.Tests.Crucible/CrucibleDataSource.cs]: Having DataStore as a property, leads to create all other methods as non static and [memberData] should be static so it results in a compilation error

**Pull request merged**